server/src/main/java/com/google/coffeehouse/StorageHandler.java

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Statement;
+
+/**
+* The StorageHandler class holds the query functions to get information from the database.
+*
+* @author  Lina Montes
+* @since   2020-07-06 
+*/
+public class StorageHandler {
+
+  /**
+  * Returns a string containing information about a person queried from the database.
+  * This method formats a string with a user's ID, email, nickname, and pronouns.
+  * If the user has no pronouns (NULL), or the pronouns are empty, then "No pronouns"
+  * is printed out instead.
+  *
+  * @param  dbClient    the database client 
+  * @param  userId      the user ID string used to query and get a person's information
+  * @return personInfo  the formatted string containing the person information
+  */
+  public static String getPersonQuery(DatabaseClient dbClient, String userId) {
+    String personInfo = "";
+    String email = "";
+    String nickname = "";
+    String pronouns = "";
+    Statement statement = 
+        Statement.newBuilder(
+                "SELECT userId, email, nickname, pronouns "
+                  + "FROM Persons "
+                  + "WHERE userId = @userId")
+              .bind("userId")
+              .to(userId)
+              .build();
+    try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
+      while (resultSet.next()) {
+        email = resultSet.getString("email");
+        nickname = resultSet.getString("nickname");
+        pronouns = "No pronouns";
+        try {
+          if (!resultSet.getString("pronouns").isEmpty()) {
+            pronouns = "Pronouns: " + resultSet.getString("pronouns");
+          }
+        } catch (Exception e) {
+          continue;
+        }
```

Comment:
What does this catch block do? It seems as though 'continue;' is unneeded as this is the last thing in the while loop.

Reply:
I might be mistaken on this (so feel free to correct me) but in instances where pronouns is null, the try will cause an exception which needs to be caught by the catch.

Reply:
You're correct that a NullPointerException will be thrown when pronouns is null, but only when the `.isEmpty()` method is called on the null result. The `continue;` in the catch block is unneeded, as all it will do is proceed to the next iteration of the while loop (which would already happen as it is the last thing in the while loop). You could simply remove the continue from the catch block, leaving it empty, and the code would work fine, but in general an empty catch block often signifies something that could be rewritten. I suggest making use `.wasNull()` method on the resultSet to first see if there were any pronouns before attempting the `.isEmpty()` call (thus, removing the need for the try catch).

```
pronouns = resultSet.getString("pronouns");
if (resultSet.wasNull() || pronouns.isEmpty()) {
  pronouns = "No pronouns";
}

```